src/main/java/com/commercetools/sync/inventory/InventorySync.java

```diff
@@ -0,0 +1,380 @@
+package com.commercetools.sync.inventory;
+
+import com.commercetools.sync.commons.BaseSync;
+import com.commercetools.sync.inventory.helpers.InventorySyncStatistics;
+import com.commercetools.sync.inventory.utils.InventorySyncUtils;
+import com.commercetools.sync.services.TypeService;
+import com.commercetools.sync.services.impl.TypeServiceImpl;
+import io.sphere.sdk.channels.Channel;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.inventory.InventoryEntry;
+import io.sphere.sdk.inventory.InventoryEntryDraft;
+import io.sphere.sdk.inventory.InventoryEntryDraftBuilder;
+import io.sphere.sdk.models.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static com.commercetools.sync.inventory.utils.InventoryDraftTransformer.transformToDrafts;
+import static java.lang.String.format;
+
+/**
+ * Default implementation of inventories sync process.
+ */
+public final class InventorySync extends BaseSync<InventoryEntryDraft, InventoryEntry, InventorySyncStatistics,
+        InventorySyncOptions> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(InventorySync.class);
+
+    /*
+     * Holds threads for executing sync process in parallel. Should be instantiated when parallel processing is
+     * enabled via InventorySyncOptions.
+     */
+    private ExecutorService executorService = null;
```

Comment:
I updated the approach, and got rid of `executorService`. 100d96ff515d13f645bc086a9c10fcb46881499d